File: SOLID/code-examples/chapter-01.cs
// FILE: chapter-01-basic.cs
// Chapter 1 — What is OOP? — Basic

using System;

// A simple class representing a Dog - this is our "blueprint"
public class Dog
{
    // Properties (data/state)
    public string Name;
    public int Age;
    
    // Method (behavior)
    public void Bark()
    {
        Console.WriteLine($"{Name} says: Woof!");
    }
}

class Program
{
    static void Main()
    {
        // Creating objects (instances) from our class
        Dog myDog = new Dog();
        myDog.Name = "Buddy";
        myDog.Age = 3;
        
        Dog friendDog = new Dog();
        friendDog.Name = "Max";
        friendDog.Age = 5;
        
        // Using the objects
        Console.WriteLine($"My dog {myDog.Name} is {myDog.Age} years old");
        myDog.Bark();
        
        Console.WriteLine($"Friend's dog {friendDog.Name} is {friendDog.Age} years old");
        friendDog.Bark();
    }
}

File: SOLID/code-examples/chapter-01.cpp
// FILE: chapter-01-basic.cpp
// Chapter 1 — What is OOP? — Basic

#include <iostream>
#include <string>

// A simple class representing a Dog
class Dog {
public:
    // Properties (data/state)
    std::string name;
    int age;
    
    // Method (behavior)
    void bark() {
        std::cout << name << " says: Woof!" << std::endl;
    }
};

int main() {
    // Creating objects (instances) from our class
    Dog myDog;
    myDog.name = "Buddy";
    myDog.age = 3;
    
    Dog friendDog;
    friendDog.name = "Max";
    friendDog.age = 5;
    
    // Using the objects
    std::cout << "My dog " << myDog.name << " is " << myDog.age << " years old" << std::endl;
    myDog.bark();
    
    std::cout << "Friend's dog " << friendDog.name << " is " << friendDog.age << " years old" << std::endl;
    friendDog.bark();
    
    return 0;
}